src/ReportBuilder.php

```diff
@@ -0,0 +1,190 @@
+
+/**
+ * @author Boris Gorbylev <ekho@ekho.name>
+ *
+ * @internal
+ */
+final class ReportBuilder
+{
+    /** @var ReportInterface[] */
+    private $reports = array();
+
+    /** @var string */
+    private $format;
+
+    /** @var array */
+    private $changed = array();
+
+    private $options = array(
+        'isDryRun' => false,
+        'isDecoratedOutput' => false,
+        'addAppliedFixers' => false,
+        'time' => null,
+        'memory' => null,
+    );
+
+    public function registerBuiltInReports()
+    {
+        /** @var string[] $builtInReports */
+        static $builtInReports;
+
+        if (null === $builtInReports) {
+            $builtInReports = array();
+
+            /** @var SplFileInfo $file */
+            foreach (SymfonyFinder::create()->files()->name('*Report.php')->in(__DIR__.'/Report') as $file) {
+                $relativeNamespace = $file->getRelativePath();
+                $builtInReports[] = 'PhpCsFixer\\Report\\'.($relativeNamespace ? $relativeNamespace.'\\' : '').$file->getBasename('.php');
+            }
+        }
+
+        foreach ($builtInReports as $reportClass) {
+            $this->registerReport(new $reportClass());
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param ReportInterface $report
+     *
+     * @return $this
+     */
+    public function registerReport(ReportInterface $report)
+    {
+        $format = $report->getFormat();
+
+        if (isset($this->reports[$format])) {
+            throw new \UnexpectedValueException(sprintf('Report for format "%s" is already registered.', $format));
+        }
+
+        $this->reports[$format] = $report;
+
+        return $this;
+    }
+
+    /**
+     * @param bool $isDryRun
+     *
+     * @return $this
+     */
+    public function setIsDryRun($isDryRun)
+    {
+        $this->options['isDryRun'] = $isDryRun;
+
+        return $this;
+    }
+
+    /**
+     * @param bool $isDecoratedOutput
+     *
+     * @return $this
+     */
+    public function setIsDecoratedOutput($isDecoratedOutput)
+    {
+        $this->options['isDecoratedOutput'] = $isDecoratedOutput;
+
+        return $this;
+    }
+
+    /**
+     * @param bool $addAppliedFixers
+     *
+     * @return $this
+     */
+    public function setAddAppliedFixers($addAppliedFixers)
+    {
+        $this->options['addAppliedFixers'] = $addAppliedFixers;
+
+        return $this;
+    }
+
+    /**
+     * @param int $time
+     *
+     * @return $this
+     */
+    public function setTime($time)
+    {
+        $this->options['time'] = $time;
+
+        return $this;
+    }
+
+    /**
+     * @param int $memory
+     *
+     * @return $this
+     */
+    public function setMemory($memory)
+    {
+        $this->options['memory'] = $memory;
+
+        return $this;
+    }
+
+    /**
+     * @param string $format
+     *
+     * @return $this
+     */
+    public function setFormat($format)
+    {
+        $this->format = $format;
```

Comment:
but `format` used for creating concrete report in builder, but `options` passes to this concrete report